support/cas-server-support-aup-rest/src/main/java/org/apereo/cas/aup/RestAcceptableUsagePolicyRepository.java

```diff
@@ -81,21 +82,24 @@ public Optional<AcceptableUsagePolicyTerms> fetchPolicy(final RequestContext req
             val rest = aupProperties.getRest();
             val url = StringUtils.appendIfMissing(rest.getUrl(), "/").concat("policy");
             val principal = WebUtils.getAuthentication(requestContext).getPrincipal();
+            val request = WebUtils.getHttpServletRequestFromExternalWebflowContext(requestContext);
 
             val exec = HttpUtils.HttpExecutionRequest.builder()
                 .basicAuthPassword(rest.getBasicAuthPassword())
                 .basicAuthUsername(rest.getBasicAuthUsername())
                 .method(HttpMethod.valueOf(rest.getMethod().toUpperCase()))
                 .url(url)
-                .parameters(CollectionUtils.wrap("username", principal.getId()))
+                .parameters(CollectionUtils.wrap("username", principal.getId(),
+                    "locale", request.getLocale().toString()))
                 .build();
             response = HttpUtils.execute(exec);
             val statusCode = response.getStatusLine().getStatusCode();
+            val result = IOUtils.toString(response.getEntity().getContent(), StandardCharsets.UTF_8);
             if (HttpStatus.valueOf(statusCode).is2xxSuccessful()) {
-                val result = IOUtils.toString(response.getEntity().getContent(), StandardCharsets.UTF_8);
                 val terms = MAPPER.readValue(JsonValue.readHjson(result).toString(), AcceptableUsagePolicyTerms.class);
                 return Optional.ofNullable(terms);
             }
+            LOGGER.debug("AUP fetch policy request returned with response code [{}] and content [{}]", statusCode, result);
```

Comment:
Sorry about the back and forth; this one should be a WARN.

Reply:
no problem